File: mcap-quartz/src/main/java/com/marshal/mcap/quartz/service/QuartzService.java
package com.marshal.mcap.quartz.service;

import com.marshal.mcap.quartz.entity.JobCreateInfo;
import com.marshal.mcap.quartz.entity.JobDetails;
import org.quartz.SchedulerException;

import java.util.List;
import java.util.Map;

/**
 * Created with IntelliJ IDEA.
 * author: Marshal
 * Date: 2018/11/1
 * Time: 20:34
 * Description:
 */
public interface QuartzService {

    /**
     *scheduler方法
     */
    boolean startup() throws SchedulerException;

    boolean standby() throws SchedulerException;

    Map<String, Object> schedulerInformation() throws SchedulerException;

    /**
     *job方法
     */
    List<JobDetails> queryJobDetail(JobDetails condition, int pageNum, int pageSize);

    void pauseJob(String jobName, String jobGroup) throws SchedulerException;

    void resumeJob(String jobName, String jobGroup) throws SchedulerException;

    void createJob(JobCreateInfo jobCreateInfo)throws Exception;

    void deleteJob(String jobName, String jobGroup) throws SchedulerException;
 }

File: halcyon-web/src/main/resources/static/js/service/quartz/quartzService.js
/**
 * author:Marshal
 * description:service层,与数据库交互
 */
app.service("quartzService",function($http){
    /**
     * scheduler方法
     */
    this.startUp=function () {
        return $http.get("../quartz/scheduler/startup");
    }
    this.standBy=function () {
        return $http.get("../quartz/scheduler/standby");
    }


    /**
     *job方法
     */
    this.query=function(pageNum,pageSize,condition){
        return $http.post("../quartz/jobDetail/query?pageNum="+pageNum+"&pageSize="+pageSize,condition);
    }

    this.pauseJob=function (jobName,jobGroup) {
        return $http.get("../quartz/job/pauseJob?jobName="+jobName+"&jobGroup="+jobGroup);
    }

    this.resumeJob=function (jobName,jobGroup) {
        return $http.get("../quartz/job/resumeJob?jobName="+jobName+"&jobGroup="+jobGroup);
    }

    this.createJob=function (jobCreateInfo) {
        return $http.post("../quartz/job/createJob",jobCreateInfo);
    }

    this.deleteJob=function (jobName,jobGroup) {
        return $http.get("../quartz/job/deleteJob?jobName="+jobName+"&jobGroup="+jobGroup);
    }
});